dotCMS/src/main/java/com/dotmarketing/portlets/rules/util/RulesImportExportUtil.java

```diff
@@ -83,69 +68,69 @@ public void export(File file) throws IOException {
+
+		try (BufferedReader bufferedReader = Files.newBufferedReader(Paths.get(file.toURI()))) {
 
-		BufferedReader in = null;
-		try {
-			User user = APILocator.getUserAPI().getSystemUser();
-			FileReader fstream = new FileReader(file);
-			in = new BufferedReader(fstream);
-			ObjectMapper mapper = new ObjectMapper();
 			mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
-			StringWriter sw = new StringWriter();
+
 			String str;
-			while ((str = in.readLine()) != null) {
-				sw.append(str);
+			while ((str = bufferedReader.readLine()) != null) {
+				stringWriter.append(str);
 			}
 
-			RulesImportExportObject importer = mapper.readValue(sw.toString(), RulesImportExportObject.class);
+			RulesImportExportObject importer = mapper.readValue
+					((String) stringWriter.toString(), RulesImportExportObject.class);
+
+			this.importRules(importer, APILocator.systemUser());
+		} catch (Exception e) {
+			Logger.error(this.getClass(), "Error: " + e.getMessage(), e);
+		}
+	}
+
+	@WrapInTransaction
+	public void importRules(final RulesImportExportObject importer,
+			final User user) throws IOException, DotDataException {
+
+		try {
+			final RulesAPI rulesAPI = APILocator.getRulesAPI();
 
 			//Saving the rules.
 			for (Rule rule : importer.getRules()) {
 				rule.setParentPermissionable(RulePermissionableUtil.findParentPermissionable(rule.getParent()));
-				rapi.saveRule(rule, user, false);
+				rulesAPI.saveRule(rule, user, false);
 
 				//Saving the Condition Groups.
 				for (ConditionGroup group : rule.getGroups()) {
-					rapi.saveConditionGroup(group, user, false);
+					rulesAPI.saveConditionGroup(group, user, false);
 					//Saving the Condition for each group.
 					for (Condition condition : group.getConditions()) {
-						rapi.saveCondition(condition, user, false);
+						rulesAPI.saveCondition(condition, user, false);
 					}
 				}
 				//Saving the Action.
 				for (RuleAction action : rule.getRuleActions()) {
-					rapi.saveRuleAction(action, user, false);
+					rulesAPI.saveRuleAction(action, user, false);
 				}
 			}
-
 		} catch (Exception e) {
 			Logger.error(this.getClass(), "Error: " + e.getMessage(), e);
-		} finally {
-			in.close();
+			throw new DotDataException(e);
 		}
 	}
 
-	/**
-	 * 
-	 * @param parent
-	 * @return
-	 * @throws DotDataException
-	 * @throws DotSecurityException
-	 */
-	public RulesImportExportObject buildExportObject(Ruleable parent) throws DotDataException, DotSecurityException {
+	@CloseDBIfOpened
+	public RulesImportExportObject buildExportObject(Ruleable parent)
+			throws DotDataException, DotSecurityException {
 
-		RulesAPI rapi = APILocator.getRulesAPI();
+		RulesAPI rulesAPI = APILocator.getRulesAPI();
 		User user = APILocator.getUserAPI().getSystemUser();
 
-		List<Rule> rules = (parent == null) ? rapi.getAllRules(user, false) : rapi.getAllRulesByParent(parent, user, false);
+		List<Rule> rules = (parent == null) ? rulesAPI.getAllRules(user, false)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'rules' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4925354)